lib/Dialect/TTNN/Analysis/LegalGridAnalysis.cpp

```diff
@@ -11,7 +11,13 @@ namespace mlir::tt::ttnn {
 
 bool mock_is_output_tensor_legal_for_op(Operation *op, tt::LayoutAttr layout) {
   // Placeholder, needs to be replaced with a call the the TTNN op interface.
-  return true;
+
+  if (TTNNOpBackend backend = dyn_cast<TTNNOpBackend>(op)) {
+    return backend.isOpLegal(layout);
```

Comment:
So my understanding is that we want to solve constraint check in shard solver first, then in legal grid analysis. The difference being that in shard solver we have the inputs as well as the outputs, so the signature should look something like this:
```
bool does_op_support_input_and_output_layout(Operation* op, LayoutAttr output_tensor_layout, vector<LayoutAttr> input_tensor_layouts) {
    …
}
```

Reply:
good call. per offline discussion adding "const std::vector<tt::LayoutAttr>&":$input_layouts as another arguments and moving this call to bool ShardSolver::checkShardCompatible